File: LLM-Translations/GPT-4/codenet/C++/Go/s410514868.go

package main

import (
	"fmt"
)

func main() {
	var n, k, ans, L, ans2 int64
	var cnt int64
	var s string

	fmt.Scan(&n, &k)
	fmt.Scan(&s)

	L = int64(s[0])

	for i := 1; i < len(s); i++ {
		if L == int64(s[i]) {
			cnt++
		} else {
			L = int64(s[i])
			ans += cnt
			ans2++
			cnt = 0
		}
	}
	ans += cnt
	ans2++

	fmt.Println(ans + min(2*k, ans2-1))
}

func min(a, b int64) int64 {
	if a < b {
		return a
	}
	return b
}

// 

File: LLM-Translations/GPT-4/codenet/C++/Java/s410514868.java

import java.util.Scanner;

public class s410514868 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        long n = scanner.nextLong();
        long k = scanner.nextLong();
        String s = scanner.next();
        long ans = 0, L, ans2 = 0, cnt = 0;
        L = s.charAt(0);
        for (int i = 1; i < n; i++) {
            if (L == s.charAt(i)) {
                cnt++;
            } else {
                L = s.charAt(i);
                ans += cnt;
                ans2++;
                cnt = 0;
            }
        }
        ans += cnt;
        ans2++;
        System.out.println(ans + Math.min(2 * k, ans2 - 1));
    }
}


